source/pci/pci.go

```diff
@@ -106,10 +110,16 @@ func (s Source) Discover() (source.Features, error) {
 func readDevInfo(devPath string) (pciDeviceInfo, error) {
 	info := pciDeviceInfo{}
 
-	for _, attr := range devLabelAttrs {
+	for _, attr := range append(devLabelAttrs, extraDevAttrs...) {
 		data, err := ioutil.ReadFile(path.Join(devPath, attr))
 		if err != nil {
-			return info, fmt.Errorf("Failed to read device %s: %s", attr, err)
+			// extraDevAttrs are considered optional so we ignore errors and continue
+			for _, optional := range extraDevAttrs {
+				if strings.Compare(attr, optional) != 0 {
+					return info, fmt.Errorf("Failed to read device %s: %s", attr, err)
+				}
+			}
```

Comment:
This logic looks like it breaks if there are more than one item in `extraDevAttrs`.

How about creating a `map[string]struct{}` of the optional keys and then doing a lookup in the for loop? Another (better?) solution would be to change give the list of attrs (plus a bool flag if to fail on error) as an argument to `readDevInfo()` and calling it twice, first with the mandatory and then with the extra attrs.